### DISCUSSION TITLE
Swizzle with Context Provider?

### DISCUSSION THREAD

**Discussion by AndrewSepic** [facebook/docusaurus#9923] [Q&A] [answered]
I've created a swizzle of the theme-classic `DocCardList` [component](https://github.com/facebook/docusaurus/blob/319d73d54f3ab3bc0d2294ed0539088fb0aba135/packages/docusaurus-theme-classic/src/theme/DocCardList/index.tsx).  But instead of using `useCurrentSidebarCategory();` to generate the items for the list, I want this to exist at the top level of my docs site.  ie:  `src/docs/`. No folder within `/docs`, no categories etc..  

I've swizzled the DocCardList and attempted to use `useDocsSidebar()` which throws an error that it's outside of the `DocSidebarProvider`.  I swizzled a `theme/Root.js` component to add the provider 

```
import { DocsSidebarProvider } from '@docusaurus/theme-common/internal';

// Default implementation, that you can customize
export default function Root({ children }) {
  return <DocsSidebarProvider>{children}</DocsSidebarProvider>;
}
```

While the error is no longer thrown my`useDocsSidebar()` call returns null when trying to access the sidebar data in the swizzled `DocCardList` component.   

Is it possible to Swizzle a Context Provider?  Is there some logic here that I'm missing?  

Lastly, is there an easier way to accomplish my goal of building a DocCardList based on the top level pages (ie: there are no categories or folders in `src/docs/`.

**Comment by slorber** [comment]
You don't have to use context/provider for this, the component accepts props too. Just give it an hardcoded array of items as props and it will work.

```jsx
<DocCardList items={[]}/>
```

Note: this is not publicly documented so we can't guarantee the shape of the items objects will remain retrocompatible forever.

---

Using `<DocCardList/>` at a doc at the root of your docs sidebar should be permitted. If we don't support that, please open a bug report with a docusaurus.new repro, and we'll look into supporting that.

**Reply by AndrewSepic** [reply]
Thanks for the reply.  I'm aware the component accepts props as well, and happy to use props but I'd like to retrieve the array of sidebar items from my markdown pages at `src/docs/ ___`. And I'm not aware of how to do that without using `useDocsSidebar()` either above `DocCardList` or within it.  The default implementation of `DocCardList` uses `useCurrentSidebarCategory();` to generate those pages and will not work at the docs root.

**Reply by slorber** [reply, marked as answer]
Please open a bug report and we'll add automatic support for this

This temporary workaround should work:

```js
import DocCardList from '@theme/DocCardList';
import { useDocsSidebar } from '@docusaurus/theme-common/internal';

<DocCardList items={useDocsSidebar().items} />
```

https://stackblitz.com/edit/github-mu7uz9?file=docs%2Fintro.mdx

![CleanShot 2024-03-07 at 18 50 47](https://github.com/facebook/docusaurus/assets/749374/70f651a2-3807-4ef5-941b-810c090e8e15)


The above sandbox should work with just `<DocCardList />`, but currently it doesn't work: this is a bug.

**Reply by AndrewSepic** [reply]
@slorber The work around works nicely in my use case.  I will open a bug report!  I appreciate it.

**Reply by slorber** [reply]
Will be fixed in https://github.com/facebook/docusaurus/pull/10847
